### PR TITLE
Change firebase implementation for OrderServlet and MyOrderServlet to use DI

### PR DESCRIPTION
**Pull request by MichaelaTSH**

### CODE REVIEW

File: webproject/src/main/webapp/js/room-my-order.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ window.onload = function() {
 async function getMyOrder() {
     let orderResponse = await fetch(`/myOrder?roomId=${roomID}`);
     let myOrders = await orderResponse.json();
-    let myOrderItems = Object.values(myOrders);
+    let myOrderItems = Object.values(myOrders)[0];
```
Comment (kevinluvian) [inline webproject/src/main/webapp/js/room-my-order.js]: what does this for?

Comment (kevinluvian) [reply webproject/src/main/webapp/js/room-my-order.js]: please use myOrders.orders

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/webapp/js/room-my-order.js (hunk 1 above)
Comment (MichaelaTSH) [reply webproject/src/main/webapp/js/room-my-order.js]: Thank you, I didn't realise I could do that! I've changed it.

File: webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        Optional<List<DataSnapshot>> dataSnapshots = firebaseUtil.getAllQuerySnapshots(query, userEmailRoomId);
+
+        List<Order> orders = new ArrayList<>();
+
+        if (!dataSnapshots.isPresent()) {
+            return null;
+        }
+
+        for (DataSnapshot dataSnapshot : dataSnapshots.get()) {
+            String product = dataSnapshot.child("product").getValue(String.class);
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java]: move this to Order class, create a function called loadFromDataSnapshot.

Imagine order class get a new attribute, how does the person know that they need to change this line as well?

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java (hunk 2 above)
Comment (MichaelaTSH) [reply webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java]: I'm currently using an OrderProto which I'm not sure I can edit. Would it be ok to make that function in the OrderManager class instead?

File: webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    public List<Order> getMyOrders(String roomId, String userEmail) throws ServletException {
+        String userEmailRoomId = userEmail + "_" + roomId;
+        Query query = firebaseUtil.getOrdersReference().orderByChild("userEmailRoomId");
+        Optional<List<DataSnapshot>> dataSnapshots = firebaseUtil.getAllQuerySnapshots(query, userEmailRoomId);
+
+        if (!dataSnapshots.isPresent()) {
+            return null;
+        }
+
+        List<Order> orders = dataSnapshots.get().stream().map(this::toOrder).collect(Collectors.toList());
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java]: return directly

**Review by kevinluvian — CHANGES_REQUESTED** [review]

**Review by kevinluvian — APPROVED** [review]
you can merge after you fix it

File: webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java (hunk 3 above)
Comment (MichaelaTSH) [reply webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java]: Done!

**Pull request merged**